arch/eventloop_posix_udp.c

```diff
@@ -36,6 +44,524 @@ typedef struct {
+    if(error != 0) {
+        UA_LOG_SOCKET_ERRNO_GAI_WRAP(
+            UA_LOG_WARNING(logger, UA_LOGCATEGORY_NETWORK,
+                           "UDP\t| Lookup of %s failed with error %d - %s",
+                           hostname, error, errno_str));
+        return -1;
+    }
+    return 1;
+}
+static UA_INLINE UA_StatusCode
+setLoopBackData(UA_SOCKET sockfd,
+                UA_Boolean enableLoopback,
+                int ai_family,
+                const UA_Logger *logger) {
+    /* Set loop back data to your host */
+    /* The Linux Kernel IPv6 socket code checks for optlen to be at least the
+     * size of an integer. However, channelDataUDPMC->enableLoopback is a
+     * boolean. In order for the code to work for IPv4 and IPv6 propagate it to
+     * a temporary integer here. */
+    UA_Int32 enable = enableLoopback;
+#if UA_IPV6
+    if(UA_setsockopt(sockfd,
+                     ai_family == AF_INET6 ? IPPROTO_IPV6 : IPPROTO_IP,
+                     ai_family == AF_INET6 ? IPV6_MULTICAST_LOOP : IP_MULTICAST_LOOP,
+                     (const char *)&enable,
+                     sizeof (enable)) < 0)
+#else
+        if(UA_setsockopt(sockfd, IPPROTO_IP, IP_MULTICAST_LOOP,
+                     (const char *)&enable,
+                     sizeof (enable)) < 0)
+#endif
+    {
+        UA_LOG_SOCKET_ERRNO_WRAP(
+            UA_LOG_ERROR(logger, UA_LOGCATEGORY_NETWORK,
+                         "PubSub Connection creation failed. Loopback setup failed: "
+                         "Cannot set socket option IP_MULTICAST_LOOP. Error: %s",
+                         errno_str));
+        return UA_STATUSCODE_BADINTERNALERROR;
+    }
+    return UA_STATUSCODE_GOOD;
+}
+
+static UA_INLINE void
+setTimeToLive(UA_SOCKET sockfd,
+              UA_UInt32 messageTTL,
+              int ai_family,
+              const UA_Logger *logger) {
+    /* Set Time to live (TTL). Value of 1 prevent forward beyond the local network. */
+#if UA_IPV6
+    if(UA_setsockopt(sockfd,
+                     ai_family == PF_INET6 ? IPPROTO_IPV6 : IPPROTO_IP,
+                     ai_family == PF_INET6 ? IPV6_MULTICAST_HOPS : IP_MULTICAST_TTL,
+                     (const char *)&messageTTL,
+                     sizeof(messageTTL)) < 0)
+#else
+        if(UA_setsockopt(sockfd, IPPROTO_IP, IP_MULTICAST_TTL,
+                     (const char *)&messageTTL,
+                     sizeof(messageTTL)) < 0)
+#endif
+    {
+        UA_LOG_SOCKET_ERRNO_WRAP(
+            UA_LOG_WARNING(logger, UA_LOGCATEGORY_NETWORK,
+                           "PubSub Connection creation problem. Time to live setup failed: "
+                           "Cannot set socket option IP_MULTICAST_TTL. Error: %s",
+                           errno_str));
+    }
+}
+
+static UA_INLINE void
+setReuseAddress(UA_SOCKET sockfd, UA_Boolean enableReuse, const UA_Logger *logger) {
+    /* Set reuse address -> enables sharing of the same listening address on
+    * different sockets */
+    if(enableReuse){
```

Comment:
Early return instead of wrapping everything into an if clause.
Keep the code flat.

Reply:
> Early return instead of wrapping everything into an if clause. Keep the code flat.

Good for pointing this out, there was a bug anyway: if (enableReuse) should just set the reuseValue to 1 else it is 0 and the setting of the option have to be done either way.